Packages/com.unity.inputsystem/InputSystem/Editor/Analytics/InputActionCodeAuthoringAnalytic.cs

```diff
@@ -0,0 +1,168 @@
+#if UNITY_EDITOR
+
+using System;
+using UnityEditor;
+using UnityEngine.Serialization;
+
+namespace UnityEngine.InputSystem.Editor
+{
+#if UNITY_2023_2_OR_NEWER
+    [UnityEngine.Analytics.AnalyticInfo(eventName: kEventName, maxEventsPerHour: kMaxEventsPerHour,
+        maxNumberOfElements: kMaxNumberOfElements, vendorKey: UnityEngine.InputSystem.InputAnalytics.kVendorKey)]
+#endif // UNITY_2023_2_OR_NEWER
+    internal class InputActionCodeAuthoringAnalytic : UnityEngine.InputSystem.InputAnalytics.IInputAnalytic
+    {
+        public const string kEventName = "input_action_code_authoring";
+        public const int kMaxEventsPerHour = 100; // default: 1000
+        public const int kMaxNumberOfElements = 100; // default: 1000
+
+        /// <summary>
+        /// Enumeration type for code authoring APIs mapping to <see cref="InputActionSetupExtensions"/>.
+        /// </summary>
+        /// <remarks>
+        /// This enumeration type may be added to, but NEVER changed, since it would break older data.
+        /// </remarks>
+        public enum Api
+        {
+            AddBinding = 0,
+            AddCompositeBinding = 1,
+            ChangeBinding = 2,
+            ChangeCompositeBinding = 3,
+            Rename = 4,
+            AddControlScheme = 5,
+            RemoveControlScheme = 6,
+            ControlSchemeWithBindingGroup = 7,
+            ControlSchemeWithDevice = 8,
+            ControlSchemeWithRequiredDevice = 9,
+            ControlSchemeWithOptionalDevice = 10,
+            ControlSchemeOrWithRequiredDevice = 11,
+            ControlSchemeOrWithOptionalDevice = 12
```

Comment:
Yes they are, but while I was going at it I had to carefully make sure each one was properly mapped through all the overloads and calls between them so I thought it made sense to keep the detail and throw it away in the end instead in case we need it later it would be easy to switch.